src/main/java/com/scalyr/integrations/kafka/AddEventsClient.java

```diff
@@ -154,7 +155,14 @@ public AddEventsClient(String scalyrUrl, String apiKey, long addEventsTimeoutMs,
         .setEvents(events);
 
       outputStream.reset();
-      addEventsRequest.writeJson(compressor.newStreamCompressor(outputStream));
+
+      // NOTE: We use countingStream since we also need access to the raw serialized payload size before the compression
+      OutputStream compressorStream = compressor.newStreamCompressor(outputStream);
```

Comment:
I'm not a Java expert so not sure if that's the best and most efficient way to get access to this data.

Based on my research, if the underlying output stream doesn't expose this data to us (aka number of bytes it read), we don't have many other options but to use ``CountingOutputStream`` or similar.

Reply:
Yes, this is a good way and how we do it on the server side also.